waiter/src/waiter/token.clj

```diff
@@ -24,13 +24,26 @@
 (def ^:const ANY-USER "*")
 (def ^:const valid-token-re #"[a-zA-Z]([a-zA-Z0-9\-_$\.])+")
 
+(defn sanitize-history
+  "Limits the history length stored in the token-data."
+  [history-length token-data]
+  (utils/dissoc-in token-data (repeat history-length "previous")))
+
+(defn ensure-history
+  "Ensures a non-nil previous entry exists in `token-data`.
+   If one already was present when this function was called, returns `token-data` unmodified.
+   Else if assocs `(or previous {})` into `token-data`."
+  [previous token-data]
```

Comment:
Should we have `token-data` since it's the most important argument?

Reply:
Placing the argument second was motivated by the use of `->>` in the locations it is used.